python/cudf/cudf/io/csv.py

```diff
@@ -58,6 +59,14 @@ def read_csv(
     if na_values is not None and is_scalar(na_values):
         na_values = [na_values]
 
+    if keep_default_na is False:
+        # TODO: Remove this error once the following issue is fixed:
+        # https://github.com/rapidsai/cudf/issues/6680
+        raise NotImplementedError(
+            "keep_default_na=False is currently not supported, please refer "
+            "to: https://github.com/rapidsai/cudf/issues/6680"
+        )
+
```

Comment:
I'd like to keep #6680 out of this PR. I'm planning to address #6680 and #6682 soon( ~1 day of work). If it turns out to be a bigger issue, I'm okay with this workaround.

Reply:
@vuule it's super easy to delete this block of code once a fix lands and protects us a bit until the fix lands. Any opposition?

Reply:
My concern was that my PR and this one would be open at the same time. Since this already looks ready to merge, I'm okay with merging it with this block.